docs/dynamic-component-loading.md

```diff
@@ -69,10 +69,19 @@ The second overload adds the ability to specify:
 * The new load context is initialized by specifying the full path to the main assembly of the component to load.  
 * It will look for the `.deps.json` next to that assembly to determine its dependencies. Lack of `.deps.json` will be treated in the same way it is today for executable apps - that is all the assemblies in the same folder will be used as dependencies.
 * Parsing and understanding of the `.deps.json` will be performed by the same host components which do this for executable apps (so same behavior/quirks/bugs, very little code duplication).
-* If the component has `.runtimeconfig.json` it will only be used to verify runtime version and provide probing paths.
+* If the component has `.runtimeconfig.json` and/or `.runtimeconfig.dev.json` it will only be used to verify runtime version and provide probing paths.
 * The load context will determine a list of assemblies similar to TPA and list of native search paths. These will be used to resolve any assembly binding events in the load context.
 * If the load context can't resolve an assembly bind event, it will fallback to the parent load context (the app)
 
+## Handling of various asset types
+What happens with various asset types once the ALC decides to load it in isolation.
+* Normal managed assembly (code) - The `.deps.json` parsing code will return list of full file paths. The ALC will just find it there and load it.
+* Satellite assemblies (resources) - Two possibilities:
+    * Imitate app behavior exactly - `.deps.json` only provides list of resource probing paths. ALC would then try to find the `<culture>/AssemblyName.dll` in each probing path and resolve the first match.
+    * Use full paths - `.deps.json` resolution actually internally produces a list of full file paths and then trims it to just probing paths. The full file paths could be used by the ALC in a very similar manner to code assemblies.
+* Native libraries - To integrate well the ALC would only get list of native probing paths from the `.deps.json`. It would then use new API to load native library given a probing path and a simple name. Internally this would call into the existing runtime behavior (which tries various prefix/suffix combinations and so on).
+* R2R Images (NI images) - The same list of full paths as for managed assemblies may contain NI images (recognized by file extension). NI images take priority over IL images always. ALC will load the NI image.
```

Comment:
We are not using the `.ni.dll` suffix for native images. I think it was always the case in .NET Core. There are historic places that still use it, but I do not think we are shipping with the `.ni.dll` extension anywhere.

The problem with using `.ni.dll` suffix for native images is that everything has to be aware of it and strip/add it as necessary.

Reply:
I was looking into the runtime. The TPA binder will look for `.ni.dll`, `.ni.exe` and `.ni.winmd` and prioritize those over non-NI paths. Also looking into `AssemblyBinder::GetAssembly` it does make a lot of different decisions if the assembly is NI or not. And as far as I can tell, it only determines that by the `fExplicitBindToNativeImage` parameter which is set to TRUE based on the file extensions as mentioned above.

I admit I haven't debugged through it to see exactly what's going on... but it looks different to what you describe. But maybe I'm just reading the code in the wrong way.

Reply:
I agree that there are places that handle the `.ni.dll` extension. I guess I am trying to say that those paths are not exercised and likely buggy because of we are not shipping binaries with `.ni.dll` extension.

Reply:
Got it - that simplifies things.